src/lib/gundb/UserStorageClass.js

```diff
@@ -1052,76 +1072,94 @@ export class UserStorage {
       .get('byid')
       .get(event.id)
       .secretAck(event)
+      .catch(e => {
+        logger.error('writeFeedEvent failed:', e.message, e, { event })
+        throw e
+      })
   }
 
   /**
    * Subscribes to changes on the event index of day to number of events
    * the "false" (see gundb docs) passed is so we get the complete 'index' on every change and not just the day that changed
    */
   async initFeed() {
-    //load unencrypted feed from cache
-    const loadFeedCache = AsyncStorage.getItem('GD_feed')
-      .then(JSON.parse)
-      .catch(e => logger.warn('failed parsing feed from cache'))
     const { feed } = await this.gunuser
+
     logger.debug('init feed', { feed })
 
     if (feed == null) {
-      //for some reason this breaks on gun 2020 https://github.com/amark/gun/issues/987
-      await this.feed.putAck({ initialized: true }) //restore old feed data - after nullified
+      // for some reason this breaks on gun 2020 https://github.com/amark/gun/issues/987
+      await this.feed
+        .putAck({ initialized: true }) // restore old feed data - after nullified
+        .catch(e => {
+          logger.error('restore old feed data failed:', e.message, e)
+          throw e
+        })
+
       logger.debug('init empty feed', { feed })
     }
+
     this.feed.get('index').on(this.updateFeedIndex, false)
-    this.feedIds = (await loadFeedCache) || {}
 
-    // this.feed = this.gunuser.get('feed')
+    // load unencrypted feed from cache
+    this.feedIds = await AsyncStorage.getItem('GD_feed')
+      .then(JSON.parse)
```

Comment:
I suppose that JSON.parse is not necessary with the new AsyncStorage implementation.
```suggestion
```

Reply:
@MathiasMoralesDixtra 
yes, you're right. so you could immediately commit suggestions if you'll see somewhere unneeded JSON.parse (they could come from the master)